### ISSUE TITLE
Extra byte at end of MOM attribute value

### ISSUE THREAD

**Issue by timpokorny** [openlvc/portico#105]
Summary
----
_Reported by David Stratton of Distributed Simulation Lab, University of Ballarat._

I am reflecting `Manager.Federate.FederateHandle`

I see the bytes of the handle value but with an extra (null) byte afterwards

This is not the case with a normal reflection (ie not MOM)

Currently, MOM attributes are encoded as Strings using the old EncodingHelpers, which from memory stuffs a null on the end. The following method should decode it to a String happily enough for you:

`hla.rti.jlc.EncodingHelpers.decodeString( byte[] value )`

1.3 uses null-terminates strings, 1516 does not.

**Comment by michaelrfraser** [comment]
Found the code to MomFederate.getFederateHandle() here https://github.com/openlvc/portico/blob/master/codebase/src/java/portico/org/portico/lrc/services/mom/data/MomFederate.java#L60

It's calling JEncodingHelpers.encodeString(String) which is implemented as such:

```java
public static byte[] encodeString( String str )
{
	return (str + "\0").getBytes();
}
```

From what I understand from the annotation in the bug report, the null character should not be inserted if the target federate is using the 1516 interface?

I'm happy to have a look at this issue if I can get some direction on how to approach it :D

**Comment by timpokorny** [comment]
@michaelrfraser - Yes, I believe you are right, the null should be in `1.3` but not `1516`.

From the LRC you can get an `ISpecHelper`. This class is provided by the version-specific wrapper that has been created around the LRC. You can ask it which version of the HLA interface it is using. From that you could perhaps call `encodeString()` or perhaps a new `encodeString1516()` method on the `JEncodingHelpers` class that would behave properly. I'm just not sure off the top of my head whether you'll be able to get back to the LRC from within the `MomFederate`. That might take some digging around.

**Comment by timpokorny** [comment]
Will likely have to address under #55

**Comment by timpokorny** [comment]
Confirmed fixed as part of #55
